libvast/vast/legacy_type.hpp

```diff
@@ -1105,6 +1111,46 @@ auto make_inspect(caf::detail::type_list<Ts...>) {
   };
 }
 
+template <class Inspector, class... Ts>
+  requires(std::is_same<typename Inspector::result_type, bool>::value)
+auto make_inspect(caf::detail::type_list<Ts...>) {
+  return [](Inspector& f, legacy_type::inspect_helper& x) -> bool {
+    using result_type = typename Inspector::result_type;
+    if constexpr (Inspector::reads_state) {
+      if (x.type_tag != invalid_type_id)
+        caf::visit(f, x.x);
+      return true;
+    } else {
+      using reference = legacy_type&;
+      using fun = result_type (*)(Inspector&, reference);
+      static fun tbl[] = {make_inspect_fun<Inspector, Ts>()...};
+      if (x.type_tag != invalid_type_id)
+        return tbl[x.type_tag](f, x.x);
+      x.x = legacy_type{};
+      return true;
+    }
+  };
+}
+
+template <class Inspector, class... Ts>
+  requires(std::is_same<typename Inspector::result_type, void>::value)
+auto make_inspect(caf::detail::type_list<Ts...>) {
+  return [](Inspector& f, legacy_type::inspect_helper& x) -> void {
+    using result_type = typename Inspector::result_type;
+    if constexpr (Inspector::reads_state) {
+      if (x.type_tag != invalid_type_id)
+        caf::visit(f, x.x);
+    } else {
+      using reference = legacy_type&;
+      using fun = result_type (*)(Inspector&, reference);
+      static fun tbl[] = {make_inspect_fun<Inspector, Ts>()...};
+      if (x.type_tag != invalid_type_id)
+        return tbl[x.type_tag](f, x.x);
+      x.x = legacy_type{};
+    }
+  };
+}
```

Comment:
The same thing could be achieved without duplication by adding a helper lambda to the original function:
```c++
   constexpr auto ret = []{
     if constexpr (std::is_same_v<result_type, caf::error>)
       return caf::none;
     if constexpr (std::is_same_v<result_type, bool>)
       return true;
     if constexpr (std::is_same_v<result_type, void>)
       return;
   };
```
and `return ret()`.
I think I know WHY it wasn't done like that but it should be reasoned about with comment.

Reply:
I believe only the bool version will survive the upgrade to CAF 0.18. 

@dominiklohmann What do you think?

Reply:
I don't care either way. What Tobias suggested is quickly implemented, but I think it can also stay as-is. With CAF 0.18 only the `bool` version will survive as you said.

Reply:
What if I add a TODO comment to the other two version?

Reply:
Works for me. @tobim what do you think?

Reply:
I'm fine with it as long as it is explained with a comment.

Reply:
I will add a comment to the functions to be removed by the next push.